docs/contributing/FAQ.md

```diff
@@ -290,6 +291,31 @@ Once you have created your PR, you can call the **stdlib-bot** to perform basic
 
 To see other available bot commands, comment `/stdlib help` on your PR.
 
+<a name="auto-push"></a>
+
+## Why were many unrelated files automatically pushed to my PR when I committed my changes?
+
+When you open a pull request or push changes to your feature branch, GitHub compares your feature branch against your `develop` branch and shows all the differences. If your feature branch contains outdated or extra changes, they will appear in the PR, even if they are unrelated to your work.
+
+To fix this, ensure that your feature branch is based on your latest `develop` branch. You can do this by merging your `develop` branch into your feature branch:
```

Comment:
Another thing you can do is to do `git pull upstream develop`. Right now, you are assuming that your origin `develop` is in sync. That is likely not going to be the case for many new contributors who are not constantly syncing their local develop with the upstream.

Reply:
I might amend the flow to be

```bash
$ git checkout develop
$ git pull upstream develop
$ git push origin develop
$ git checkout feature-branch
$ git merge develop
```